### DISCUSSION TITLE
Toggle text-overflow: ellipsis; for md-list-item's headline and supporting-text

### DISCUSSION THREAD

**Discussion by hakataminamiWS** [material-components/material-web#5060] [Q&A] [answered]
I am using \<md-list-item\> component from release v1.0.0.
When I set a string that is longer than the width of the list to the headline or supporting-text of the list-item, it wraps around (in Chrome).
![スクリーンショット 2023-10-06 120508](https://github.com/material-components/material-web/assets/66231467/5152ec41-58cd-438c-bd55-249119fdfe24)

I would like to set it to `text-overflow: ellipsis;` when the string is longer than the list width, but I couldn't find a way to do so.

As a workaround, I tried setting
```
width: 100%;
white-space: nowrap;
```
to \<button\> (when type="button") inside the shadow dom of \<md-list-item\>.
This made it display like text-overflow: ellipsis;.
![スクリーンショット 2023-10-06 120838](https://github.com/material-components/material-web/assets/66231467/28dedb47-69a1-4728-8d83-22f123b0ed48)

Is there a way to toggle text-overflow: ellipsis; for md-list-item's headline and supporting-text?

Environment information:
OS: Windows 11
Browser: Chrome 117.0.5938.150（Official Build） （64 bit）
Version of md-list-item: v1.0.0

**Comment by asyncliz** [comment]
Looks like the `.list-item` CSS needs a `width: 100%`

Do you mind filing an issue? Would also appreciate a PR if you can manage!

**Reply by asyncliz** [reply]
Also using `white-space: nowrap` is the correct way to make the lines wrap or not

**Reply by hakataminamiWS** [reply]
Thanks for your reply.

I’m considering creating an issue or PR, but can I discuss something with you first?

I think it might be best to always set `width: 100%` and make white-space controllable. What do you think?

If we do that,
I think we have a few options:

Option a. Add like this `white-space: var(--md-list-item-white-space, 'normal');` to the .list-item CSS

Option b. Add a `part` attribute to the list-item HTML

Option c. Something else (any ideas?)

I’m leaning towards option a., but are there similar examples in other components?

**Reply by hakataminamiWS** [reply]
Perhaps without any code changes,
It may work by simply specifying the following in the slot.

```
display: -webkit-box;
-webkit-box-orient: vertical;
-webkit-line-clamp: 1;
overflow: hidden;
```       

I'll check it later and get back with the results.

**Reply by hakataminamiWS** [reply, marked as answer]
In conclusion, to apply the style like `text-overflow: ellipsis;` to `headline` and `supporting-text` of `<md-list-item>`, it seems that the following settings should be applied to each slots (`slot="headline", "supporting-text"`).

```
display: -webkit-box;
-webkit-box-orient: vertical;
-webkit-line-clamp: 1;
overflow: hidden;
```

example
![スクリーンショット 2023-10-11 165748](https://github.com/material-components/material-web/assets/66231467/ad83aef8-34f4-4239-91a0-351c6d0134a2)

**Reply by asyncliz** [reply]
That is correct, here's the [demo CSS to control wrapping and/or clamping](https://github.com/material-components/material-web/blob/04dcf25c4ae6bb334ed106b5f73ee1104e50d17c/labs/item/demo/stories.ts#L24).

```css
  /* Use this CSS to prevent lines from wrapping */
  .nowrap {
    white-space: nowrap;
  }

  /* Use this CSS on items to limit the number of wrapping lines */
  .clamp-lines {
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 2;
  }
```

I believe the list-item still needs a `width: 100%` added to it though.

**Reply by hakataminamiWS** [reply]
Hmmm... maybe. I'll take a closer look. Is there any difference with or without `width: 100%;`?

**Reply by asyncliz** [reply]
I don't believe so, the slots should expand to the width of the item
